### PR TITLE
file browser modal: add support of download file/folder

### PR DESCRIPTION
**Pull request by tao-xinxiu**

### CODE REVIEW

File: app/scripts/modals/file-browser-modal-controller.js [hunk 1]
```diff
@@ -240,6 +240,64 @@ angular.module('workflow-variables').controller('FileBrowserModalCtrl', function
         });
     }
 
+    $scope.downloadFile = function() {
+        var selectedElement=$("#files-tbody  tr.active").children().first();
+        if (selectedElement.length == 0) {
```
Comment (amouhoub) [inline app/scripts/modals/file-browser-modal-controller.js:245]: in JS you can use the existential test to check if a value is not null, not undefined, not equal to zero and not empty as a string.
to test if an array is not empty, you can do:
```if (selectedElement.length)```

File: app/views/modals/dataspace-file-browser.html [hunk 2]
```diff
@@ -23,6 +23,9 @@ <h5>
             <button id="upload-file-btn" title="{{'Upload a file'|translate}}" class="btn btn-white" ng-click="clickUpload()" ng-disabled="isUploading">
                 <i class="fa" ng-class="{'fa-upload':!isUploading, 'fa-spinner fa-pulse':isUploading}"></i>
             </button>
+            <button id="download-file-btn" title="{{'Download the selected file or folder'|translate}}" class="btn btn-white" ng-click="downloadFile()">
```
Comment (amouhoub) [inline app/views/modals/dataspace-file-browser.html:26]: you don't need to use an id unless you want to do some dom-manipulation (which is not recommended) or if it is a form with inputs and labels...

**Comment by amouhoub** [comment]
@tao-xinxiu sorry I came 1 minute late 😆

**Comment by tao-xinxiu** [comment]
> @tao-xinxiu sorry I came 1 minute late

Thanks a lot for the great reviews. I'll integrate them in the future PR.

**Pull request merged**